jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/DockerContextMojo.java

```diff
@@ -60,7 +62,12 @@ public void execute() throws MojoExecutionException {
 
     MavenProjectProperties mavenProjectProperties =
         MavenProjectProperties.getForProject(getProject(), mavenJibLogger, getExtraDirectory());
-    String mainClass = mavenProjectProperties.getMainClass(this);
+    String mainClass;
+    try {
+      mainClass = MainClassResolver.resolveMainClass(getMainClass(), mavenProjectProperties);
```

Comment:
nit: I'd just merge this `try` block with the one below and declare `mainClass` and assign to it in a single-line.